src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Variables.cs

```diff
@@ -0,0 +1,434 @@
+                    while (slot > 0)
+                    {
+                        depth++;
+                        var (id, index) = DeconstructSlot(slot);
+                        Debug.Assert(id == Id);
+                        slot = _variableBySlot[index].ContainingSlot;
+                    }
+                    return depth;
+                }
+            }
+
+            internal bool TryGetType(Symbol symbol, out TypeWithAnnotations type)
+            {
+                var variables = GetVariablesContainingVariable(symbol);
+                return variables._variableTypes.TryGetValue(symbol, out type);
+            }
+
+            internal void SetType(Symbol symbol, TypeWithAnnotations type)
+            {
+                var variables = GetVariablesContainingVariable(symbol);
+                Debug.Assert((object)variables == this);
+                variables._variableTypes[symbol] = type;
+            }
+
+            internal VariableIdentifier this[int slot]
+            {
+                get
+                {
+                    (int id, int index) = DeconstructSlot(slot);
+                    var variables = GetVariablesForId(id);
+                    return variables!._variableBySlot[index];
+                }
+            }
+
+            internal int NextAvailableIndex => _variableBySlot.Count;
+
+            internal int GetTotalVariableCount()
+            {
+                int fromContainer = Container?.GetTotalVariableCount() ?? 0;
+                return fromContainer + _variableSlot.Count;
+            }
+
+            internal void GetMembers(ArrayBuilder<(VariableIdentifier, int)> builder, int containingSlot)
+            {
+                (int id, int index) = DeconstructSlot(containingSlot);
+                var variables = GetVariablesForId(id)!;
+                var variableBySlot = variables._variableBySlot;
+                for (index++; index < variableBySlot.Count; index++)
+                {
+                    var variable = variableBySlot[index];
+                    if (variable.ContainingSlot == containingSlot)
+                    {
+                        builder.Add((variable, ConstructSlot(id, index)));
+                    }
+                }
+            }
+
+            private Variables GetVariablesForVariable(VariableIdentifier identifier)
+            {
+                int containingSlot = identifier.ContainingSlot;
+                if (containingSlot > 0)
+                {
+                    return GetVariablesForId(DeconstructSlot(containingSlot).Id)!;
+                }
+                return GetVariablesContainingVariable(identifier.Symbol);
+            }
+
+            internal Variables GetVariablesContainingVariable(Symbol symbol)
+            {
+                switch (symbol)
+                {
+                    case LocalSymbol:
+                    case ParameterSymbol:
+                    case MethodSymbol { MethodKind: MethodKind.LocalFunction }:
```

Comment:
>case MethodSymbol { MethodKind: MethodKind.LocalFunction }: [](start = 20, length = 59)

Presence of this case feels unexpected. Do we ever allocate a slot for a local function? Unless I am missing something, it looks like this case is added to accommodate NullableWalker to locate Variables for a method scope containing a local function. I think we should explicitly use method that is responsible to do that, the ```GetVariablesForSymbol```. And this method should probably be made private.